lib/northbound_grpc.cpp

```diff
@@ -1288,19 +1286,57 @@ static int frr_grpc_finish(void)
 static void frr_grpc_module_very_late_init(struct event *thread)
 {
 	const char *args = THIS_MODULE->load_args;
-	uint port = GRPC_DEFAULT_PORT;
+	std::string endpoint;
 
 	if (args) {
-		port = std::stoul(args);
-		if (port < 1024 || port > UINT16_MAX) {
-			flog_err(EC_LIB_GRPC_INIT,
-				 "%s: port number must be between 1025 and %d",
-				 __func__, UINT16_MAX);
-			goto error;
+		std::string arg_str(args);
+
+		if (arg_str.find(':') != std::string::npos) {
+			size_t pos = arg_str.rfind(':');
+			std::string port_str = arg_str.substr(pos + 1);
+			uint port = 0;
+			try {
+				port = std::stoul(port_str);
+			} catch (const std::exception &e) {
+				flog_err(EC_LIB_GRPC_INIT,
+					 "%s: invalid port number in host:port format: %s",
+					 __func__, e.what());
+				goto error;
+			}
+
+			if (port < 1024 || port > UINT16_MAX) {
+				flog_err(EC_LIB_GRPC_INIT,
+					 "%s: port number must be between 1025 and %d", __func__,
+					 UINT16_MAX);
+				goto error;
+			}
+			endpoint = arg_str;
+		} else {
+			if (!arg_str.empty() && arg_str[0] == ':')
+				arg_str = arg_str.substr(1);
+
+			uint port = 0;
+			try {
+				port = std::stoul(arg_str);
+			} catch (const std::exception &e) {
+				flog_err(EC_LIB_GRPC_INIT, "%s: invalid port number: %s", __func__,
+					 e.what());
+				goto error;
+			}
+
+			if (port < 1024 || port > UINT16_MAX) {
+				flog_err(EC_LIB_GRPC_INIT,
+					 "%s: port number must be between 1025 and %d", __func__,
+					 UINT16_MAX);
+				goto error;
+			}
+			endpoint = "0.0.0.0:" + arg_str;
 		}
+	} else {
```

Comment:
This is a style preference, and no change is required here; however, from old BSD style guidelines, when one has a conditional with a very large block followed by a very small block (as is the case here), it's better to reverse the blocks and put the very small block first so you can see both branches at once. So in this case you'd use:
```c
    if (!args) {
        endpoint = "0.0.0.0:" + std::to_string(GRPC_DEFAULT_PORT);
    } else {
        big block
        .
        .
        .
    }
```
by having the other way you can be looking at the first big block and you don't realize that there's an else clause b/c it's below and off your screen.